pandas/_libs/internals.pyi

```diff
@@ -59,8 +60,13 @@ class SharedBlock:
     _mgr_locs: BlockPlacement
     ndim: int
     values: ArrayLike
+    refs: BlockValuesRefs
```

Comment:
Actually, that's maybe a good question: _is_ it equivalent? 
With the Blockmanager tracking, the rule was that each DataFrame had its own unique manager, so one manager was never shared between different dataframes. I assume similarly here, we always create new Blocks for new Managers/DataFrames (potentially just shallow copies). So if a certain `values` is shared (referenced) by 3 DataFrames, there are 3 unique Block objects that hold the values? Or differently stated, the `referenced_blocks` list, it's always all weakrefs to unique blocks in there? Or can there ever be two weakrefs to the same Block instance?

Reply:
Theoretically this is possible (having two references to the same block). I'd prefer using shallow copies though, the way it is implemented tracking the new refs works automatically when you create a shallow copy, while you would have to explicitly update the references with your own block again. I think this would look a bit hacky

Reply:
My thinking was, we actually want to reference the values, e.g. same values living in different blocks, hence this name. But not too happy with it either, so not opposed to renaming. BlockRefs did not capture everything to me, that's why I ended up with this name